src/main/scala/chiseltest/formal/Formal.scala

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiseltest.formal
+
+import chisel3.Module
+import chiseltest.ChiselScalatestTester
+import chiseltest.experimental.sanitizeFileName
+import chiseltest.formal.backends.{FormalEngineAnnotation, Z3EngineAnnotation}
+import chiseltest.simulator.{Compiler, WriteVcdAnnotation}
+import firrtl.{AnnotationSeq, CircuitState}
+import firrtl.annotations.{Annotation, NoTargetAnnotation}
+import firrtl.transforms.formal.DontAssertSubmoduleAssumptionsAnnotation
+
+sealed trait FormalOp extends NoTargetAnnotation
+case class BoundedCheck(kMax: Int = -1) extends FormalOp
+
+/** Specifies how many cycles the circuit should be reset for. */
+case class ResetOption(cycles: Int = 1) extends NoTargetAnnotation {
+  require(cycles >= 0, "The number of cycles must not be negative!")
+}
+
+class FailedBoundedCheckException(val message: String, val failAt: Int) extends Exception(message)
+private[chiseltest] object FailedBoundedCheckException {
+  def apply(module: String, failAt: Int): FailedBoundedCheckException = {
+    val msg = s"[$module] found an assertion violation $failAt steps after reset!"
+    new FailedBoundedCheckException(msg, failAt)
+  }
+}
+
+/** Adds the `verify` command for formal checks to a ChiselScalatestTester */
+trait Formal { this: ChiselScalatestTester =>
+  def verify[T <: Module](dutGen: => T, annos: AnnotationSeq): Unit = {
+    val withTargetDir = addDefaultTargetDir(getTestName, annos)
+    Formal.verify(dutGen, withTargetDir)
+  }
+  private def getTestName: String = sanitizeFileName(scalaTestContext.value.get.name)
+}
```

Comment:
I personally think `Formal.verify` should not be only expose to Scalatest, maybe Formal can be a standalone trait?
IIRC `ChiselScalatestTester` only provides the default directory creating utils.

Reply:
The problem is that how you get a test name is completely different for `utest`.

Reply:
I changed it so that `Formal` only depends on having a `getTestName` function. Maybe you can figure out how to get that working with `utest`.